src/lib/common/util.ts

```diff
@@ -997,3 +998,19 @@ function serializeReplacer(_key: string, value: any) {
 
   return value;
 }
+
+export function marshalErrorToJSON(
+  error?: InternError
+): InternError | undefined {
+  if (!error) {
+    return undefined;
+  }
+  const { name, message, stack, lifecycleMethod } = error;
+
+  return {
+    name,
+    message,
+    stack,
+    ...(lifecycleMethod ? { lifecycleMethod } : {})
```

Comment:
For consistency, this method should handle other InternError fields as well, or at least those we currently care about (e.g., `showDiff`).